src/java.base/share/classes/jdk/internal/util/OperatingSystem.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package jdk.internal.util;
+
+import jdk.internal.util.OperatingSystemProps;
+import jdk.internal.vm.annotation.ForceInline;
+
+/**
+ * Enumeration of operating system types and testing for the current OS.
```

Comment:
Would it be more correct to say an enumerating of operating system names rather than types?

Reply:
It is more OS type or kind than name; there are multiple distros that are Linux, i.e. Ubuntu, Fedora, Oracle Linux, etc..
It is not equivalent to `os.name`.

Reply:
'Operating system families' seems well-established ?

https://en.wikipedia.org/wiki/Category:Operating_system_families